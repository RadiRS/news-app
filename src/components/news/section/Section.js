import React, { Component } from 'react';
import styled from 'styled-components';

export class Section extends Component {
  render() {
    return (
      <Container>
        <Content>
          <Title>Test</Title>
          <Author>Test</Author>
          <Date>Test</Date>
        </Content>
        <Image source={{ uri: 'http://lorempixel.com/640/480' }} />
      </Container>
    );
  }
}

export default Section;

const Container = styled.View``;

const Content = styled.View``;

const Image = styled.Image``;

const Title = styled.Text``;

const Author = styled.Text``;

const Date = styled.Text``;
