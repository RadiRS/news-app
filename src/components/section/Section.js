import React from 'react';
import styled from 'styled-components';
import { Fonts } from '../../themes';

const Section = props => {
  const { text } = props;

  return (
    <Container onLongPress={() => alert(text)}>
      <Text>{text}</Text>
    </Container>
  );
};

const Container = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  background: white;
  height: 60px;
  padding: 12px;
  margin: 8px;
  border-radius: 10px;
  elevation: 5;
`;

const Text = styled.Text`
  font-family: ${Fonts.type.bold};
  font-size: 16px;
  font-weight: ${Fonts.weight.large};
`;

export default Section;
