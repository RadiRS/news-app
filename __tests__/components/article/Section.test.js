import 'react-native';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import Section from '../../../src/components/article/section';

test('renders Section component with data article correctly', () => {
  const tree = renderer.create(<Section article={dummyDataArticle} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Section component with data search article correctly', () => {
  const tree = renderer
    .create(<Section searchResult article={dummySearchDataArticle} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const dummyDataArticle = {
  title: 'Testing title',
  multimedia: [],
  byline: 'Radi Rasadi',
  published_date: '2019-04-08T18:06:10-04:00'
};

const dummySearchDataArticle = {
  headline: { main: 'Testing title' },
  multimedia: [],
  byline: { original: 'Radi Rasadi' },
  pub_date: '2019-04-08T18:06:10-04:00'
};
