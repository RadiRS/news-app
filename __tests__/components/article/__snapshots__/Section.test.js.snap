// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Section component with data article correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 14,
        "elevation": 10,
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 150,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 0.7,
          "flexShrink": 1,
          "paddingLeft": 10,
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "Domine Bold",
            "fontSize": 22,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Testing title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#b8bece",
            "fontFamily": "Domine Bold",
            "fontSize": 15,
            "fontWeight": "600",
            "marginTop": 2,
          },
        ]
      }
    >
      Radi Rasadi
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#b8bece",
            "fontSize": 15,
            "fontWeight": "600",
            "marginTop": 2,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 0.3,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://lorempixel.com/640/480",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 14,
            "height": 100,
            "width": 100,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`renders Section component with data search article correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 14,
        "elevation": 10,
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 150,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 0.7,
          "flexShrink": 1,
          "paddingLeft": 10,
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "Domine Bold",
            "fontSize": 22,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Testing title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#b8bece",
            "fontFamily": "Domine Bold",
            "fontSize": 15,
            "fontWeight": "600",
            "marginTop": 2,
          },
        ]
      }
    >
      Radi Rasadi
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#b8bece",
            "fontSize": 15,
            "fontWeight": "600",
            "marginTop": 2,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 0.3,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://lorempixel.com/640/480",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 14,
            "height": 100,
            "width": 100,
          },
        ]
      }
    />
  </View>
</View>
`;
